File: configlibrary/src/main/java/com/shmedo/configlibrary/iot/interfaces/IOTResultParser.java
package com.shmedo.configlibrary.iot.interfaces;

import com.shmedo.configlibrary.iot.enums.IOTCommandType;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2020/8/31 <br/>
 * 描述：     TODO
 */
public interface IOTResultParser<T> {
    T parse(String result);

    void validate(String result);

    IOTCommandType commandType();
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/iot_cmd/IOTResultParser.kt
package com.shmedo.lib.device.base.iot_cmd

import com.shmedo.lib.device.base.iot_cmd.enums.IOTCommandType

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2023/9/22 <br/>
 * 描述：     TODO
 */

interface IOTResultParser<T> {
    fun validCheckBeforeParse(result: String): ValidationResult
    fun parse(result: String): ParseResult<T>
    fun commandType(): IOTCommandType
}
